aten/src/ATen/native/BackendSelectRegister.cpp

```diff
@@ -0,0 +1,54 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <c10/core/TensorOptions.h>
+
+// TODO:
+// 1) add all factory functions
+// 2) autogen this
+
+namespace at {
+namespace native {
+  Tensor empty_backend(IntArrayRef size, const TensorOptions & options, c10::optional<MemoryFormat> memory_format) {
+    DispatchKey key = options.computeDispatchKey();
+    static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::empty", "memory_format");
+    return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, const TensorOptions &, c10::optional<MemoryFormat>>(key, size, options, memory_format);
+  }
+
+  Tensor empty_strided_backend(IntArrayRef size, IntArrayRef stride, const TensorOptions & options) {
+    DispatchKey key = options.computeDispatchKey();
+    static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::empty_strided", "");
+    return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, IntArrayRef, const TensorOptions &>(key, size, stride, options);
+  }
+
+  Tensor empty_affine_quantized_backend(IntArrayRef size, const TensorOptions & options, double scale, int64_t zero_point, c10::optional<MemoryFormat> memory_format) {
+    DispatchKey key = options.computeDispatchKey();
+    static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::_empty_affine_quantized", "");
+    return op.callUnboxedWithDispatchKey<Tensor, IntArrayRef, const TensorOptions &, double, int64_t, c10::optional<MemoryFormat>>(key, size, options, scale, zero_point, memory_format);
+  }
+
+  Tensor empty_per_channel_affine_quantized_quantized_backend(IntArrayRef size, const Tensor & scales, const Tensor & zero_points, int64_t axis, const TensorOptions & options, c10::optional<MemoryFormat> memory_format) {
```

Comment:
why this set of operators?  I'm guessing this is supposed to be some "base" factory functions that others will build on, but I don't see how this one fits that example -- look at the CPU implementation of this function to see why.